packages/rum-core/test/common/config-service.spec.js

```diff
@@ -126,21 +126,27 @@ describe('ConfigService', function() {
     expect(userContext).toEqual({})
   })
 
-  it('should check config validity', function() {
-    var result = configService.isValid()
-    expect(result).toBe(false)
-
-    configService.setConfig({ serviceName: 'serviceName' })
-    result = configService.isValid()
-    expect(result).toBe(true)
-
-    configService.setConfig({ serverUrl: undefined })
-    result = configService.isValid()
-    expect(result).toBe(false)
+  it('should validate required config options', () => {
+    // Valid
+    const errors1 = configService.validate({ serviceName: 'name' })
+    expect(errors1.length).toEqual(0)
+
+    // Invalid required key serviceName
+    const errors2 = configService.validate({ serviceName: undefined })
+    expect(errors2).toEqual(['Missing serviceName'])
+
+    // Invalid required key serviceName & serverUrl
+    const errors3 = configService.validate({
+      serviceName: undefined,
+      serverUrl: ''
+    })
+    expect(errors3).toEqual(['Missing serviceName', 'serverUrl'])
```

Comment:
It seems kind of strange to me that the `errors` contain different kind of messages depending on how many errors there are. Maybe making the `validate` function more generic in a way that returns structured error and then the caller would be responsible to generate the error messages would be better, something like:

```
const errors = {
  missingKeys: ['serviceName', 'serverUrl'],
  invalidValues: [{
    key: 'serviceName',
    value: 'abc.def',
    allowed: 'a-z, A-Z, 0-9, _, -, <space>'
  }]
}
```

Reply:
Very good suggestion. I will change the validate function to output in this format.